packages/ui-markdown-editor/src/plugins/withText.js

```diff
@@ -1,22 +1,32 @@
-import { Editor, Node } from 'slate';
-import { insertThematicBreak, isBlockHeading } from '../utilities/toolbarHelpers';
+import { Editor, Node, Transforms } from 'slate';
+import { isBlockHeading } from '../utilities/toolbarHelpers';
 import { HR } from '../utilities/schema'
+import MatchHeadings from './MatchCases/MatchHeadings'
 
 export const withText = (editor) => {
-  // Inserts page break with dash
 	const { insertText } = editor;
 	editor.insertText = (text) => {
 		insertText(text);
     const currentNode = Node.get(editor, editor.selection.focus.path);
 		if(isBlockHeading(editor)){
 			return;
 		}
-		const firstWord = currentNode.text;
-		if(firstWord !== '---'){
+		
+		let matchingText;
+		const currentLine = currentNode.text;
+		const headingMatchCase = currentLine.match(/(^\s*)#{1,6}\s/m);
```

Comment:
Rather than running the regex twice, once here and then again inside MatchHeadings, I suggest we refactor this code to have an array of matching functions that take the `editor` and `currentNode.text` as input. 

Each matcher can then decide whether it wants to process the change and apply transformations to the editor. So, we would start with 2 matching functions, one for headings and one for page breaks.